include/log.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <ostream>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <variant>
+
+// TODO: Make this configurable through CMake?
+#define SPDLOG_ACTIVE_LEVEL SPDLOG_LEVEL_TRACE
+#include <spdlog/spdlog.h>
+
+#include <spdlog/fmt/ostr.h>
+#include <spdlog/sinks/stdout_color_sinks.h>
+
+#include "print_utils.h"
+
+#define CELERITY_SET_LOG_CTX(ctx) CELERITY_DETAIL_SET_LOG_CTX(ctx)
+
+#define CELERITY_TRACE(...) SPDLOG_TRACE(*celerity::detail::active_log_ctx + __VA_ARGS__)
```

Comment:
Prepending an arbitrary string to the format string will break if it contains characters that `fmt` interprets as placeholders.

Reply:
True, although I guess whether we want to do something about it depends on whether this is supposed to be a documented API that users can also use.

Reply:
@psalz couldn't that be solved rather easily with
```c++
#define CELERITY_TRACE(fmt_literal, ...) SPDLOG_TRACE("[{}] " fmt_literal, *active_log_ctx ,##__VA_ARGS__)
```
@BlackMark29A I'm not sure I like that idea. This would mean our stable API surface increases by the entirety of `libfmt`.

Reply:
I did not know of that `##__VA_ARGS__` trick - pretty neat! Alas, it seems like it's unsupported in MSVC.

Reply:
[Seems like it's supported in the new preprocessor though](https://docs.microsoft.com/en-us/cpp/preprocessor/preprocessor-experimental-overview?view=msvc-170#comma-elision-in-variadic-macros). So we'd have to enable that through CMake; not sure how stable it is though.

Reply:
I agree we shouldn't rely on that. Is there another way to express this, maybe similarly to how the `WORKAROUND` macros are defined right now?

Reply:
Fixed through double-formatting as suggested by @fknorr during Monday's meeting.